src/cdvdcontrol.cpp

```diff
@@ -0,0 +1,1433 @@
+                return 5;
+            }
+        }
+        else if((strcmp(argv[i],"--hcdr")) == 0) {
+            flags |= FL_HCDR;
+            if (argc>(i+1)) {
+                i++;
+                if((strcmp(argv[i],"off")) == 0) {
+                    hcdr = 0;
+                }
+                else if ((strcmp(argv[i],"on")) == 0) {
+                    hcdr = 1;
+                } else {
+                    std::cout << "Illegal argument for Hide-CDR: " << argv[i] << "\n";
+                }
+            } else {
+                needs_parameter(argv, i);
+                return 5;
+            }
+        }
+//  ************   PoweREC
+        else if((strcmp(argv[i],"--powerec")) == 0) {
+            flags |= FL_POWEREC;
+            if (argc>(i+1)) {
+                i++;
+                if     ((strcmp(argv[i],"off")) == 0) {
+                    powerec = 0;
+                }
+                else if ((strcmp(argv[i],"on")) == 0) {
+                    powerec = 1;
+                } else {
+                    std::cout << "Illegal argument for PoweRec: " << argv[i] << "\n";
+                }
+            } else {
+                needs_parameter(argv, i);
+                return 5;
+            }
+        }
+//  ************   GigaREC
+        else if((strcmp(argv[i],"--gigarec")) == 0) {
+            if (argc>(i+1)) {
+                i++;
+                flags |= FL_GIGAREC;
+                if      ((strcmp(argv[i],"0.6")) == 0) {
+                    gigarec = GIGAREC_06;
+                }
+                else if ((strcmp(argv[i],"0.7")) == 0) {
+                    gigarec = GIGAREC_07;
+                }
+                else if ((strcmp(argv[i],"0.8")) == 0) {
+                    gigarec = GIGAREC_08;
+                }
+                else if ((strcmp(argv[i],"0.9")) == 0) {
+                    gigarec = GIGAREC_09;
+                }
+                else if ((strcmp(argv[i],"1.0")) == 0 || ((strcmp(argv[i],"off")) == 0)) {
+                    gigarec = GIGAREC_10;
+                }
+                else if ((strcmp(argv[i],"1.1")) == 0) {
+                    gigarec = GIGAREC_11;
+                }
+                else if ((strcmp(argv[i],"1.2")) == 0) {
+                    gigarec = GIGAREC_12;
+                }
+                else if ((strcmp(argv[i],"1.3")) == 0) {
+                    gigarec = GIGAREC_13;
+                }
+                else if ((strcmp(argv[i],"1.4")) == 0) {
+                    gigarec = GIGAREC_14;
+                }
+                else {
+                    std::cout << "Illegal GigaREC value: " << argv[i] << "\n";
+                }
+            } else {
+                needs_parameter(argv, i);
+                return 5;
+            }
+        }
+//  ************   VariREC  CD
+        else if((strcmp(argv[i],"--varirec-cd")) == 0) {
+            if (argc>(i+1)) {
+                i++;
+                flags |= FL_VARIREC_CD;
+                int val = 0;
```

Comment:
Codacy found an issue: [The scope of the variable 'val' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)